src/main/java/com/online/market/http/validation/AgeValidator.java

```diff
@@ -1,13 +1,46 @@
 package com.online.market.http.validation;
 
+import com.sun.javaws.exceptions.InvalidArgumentException;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
 import java.util.Date;
 
 /**
  * Created by Alexander on 04.06.2017.
  */
 public class AgeValidator implements Validator<Date> {
     public Date validate(Date birthday) {
+        DateFormat dateFormat = new SimpleDateFormat("dd-MM-yyyy");
+        String birthDay = dateFormat.format(birthday);
+        String currentDate = dateFormat.format(new Date());
+
+        String partsCurrentDate[] = currentDate.split("\\-");
+        String partsBirthday[] = birthDay.split("\\-");
+
+        if (!dateCompare(partsCurrentDate, partsBirthday)) throw new InvalidArgumentException();  // need help to choose a kind of exception =)
+        else return birthday;
+
         // TODO person should be 18 years old
-        return null;
+    }
+
+    private boolean dateCompare(String currentDate[], String birthday[]) {
```

Comment:
this approach is very naive, try:
 
to use `Joda-Time` library.
For your needs you can use the Years.yearsBetween() method.